modules/apps/kate.nix

```diff
@@ -96,6 +104,56 @@ in
 
     "KTextEditor Renderer" = {
       "Show Indentation Lines" = cfg.editor.indent.showLines;
+
+
+      # COLORTHEME (cannot define this below)
+      # Do pick the theme if the user chose one,
+      # Do not touch the theme settings otherwise
+      "Auto Color Theme Selection" = lib.mkIf (cfg.editor.theme.name != "") false;
+      "Color Theme" = lib.mkIf (cfg.editor.theme.name != "") cfg.editor.theme.name;
     };
   };
+
+
+  # ==================================
+  #     COLORTHEME
+
+  options.programs.kate.editor.theme = {
+    src = lib.mkOption {
+      description = "The path of a theme file for the KDE editor (not the window color scheme). Obtain a custom one by using the GUI settings in kate. If you want to use a system-wide editor color scheme set this path to null. If you set the metadata.name entry in the file to a value that matches the name of a system-wide color scheme undesired behaviour may occur. The activation will fail if a theme with the filename `<name of your theme>.theme` already exists.";
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    name = lib.mkOption {
+      description = "The name of the theme in use. May be a system theme. If a theme file was submitted this setting will be set automatically.";
+      type = lib.types.str;
+      default = "";
+    };
+  };
+
+  config.programs.kate.editor.theme = {
+    name = lib.mkIf (cfg.enable && null != cfg.editor.theme.src) (lib.mkForce (builtins.fromJSON (builtins.readFile cfg.editor.theme.src))."metadata"."name");
+    # kate's naming scheme is ${themename}.theme
+    # which is why we use the same naming scheme here
+  };
+
+  # This won't override existing files since the home-manager activation fails in that case
+  config.xdg.dataFile."${cfg.editor.theme.name}.theme" = lib.mkIf (cfg.enable && null != cfg.editor.theme.src)
+  {
+    source = cfg.editor.theme.src;
+    target = "org.kde.syntax-highlighting/themes/${cfg.editor.theme.name}.theme";
+  };
+
+  config = {
+    home.packages = lib.mkIf (cfg.enable && cfg.package != null) [ cfg.package ];
+
+    # In case of using a custom theme, check that there is no name collision
+    home.activation.checkKateTheme = lib.mkIf (cfg.enable && cfg.editor.theme.src != null) (lib.hm.dag.entryBefore [ "writeBoundary" ]
+    ''
+      ${./check-theme-name-free.sh} ${cfg.editor.theme.name} ${pkgs.jq}/bin/jq
```

Comment:
Here I would suggest creating a nix-derivation here which runs the script. That way you don't have to send in the jquery path as a command-line argument, and in general is a better practice. See [here](https://github.com/pjones/plasma-manager/blob/d0e5698aa41e331ab36e9700b4be51f40127a47d/lib/writeconfig.nix#L4) for how we did with the config-writing script

Reply:
I changed to to a derivation in the same style. Imo it looks a bit bloated:

```nix
    checkThemeNameScript = pkgs.writeShellApplication {
      name = "checkThemeName";
      runtimeInputs = with pkgs; [ jq ];
      text = builtins.readFile ./check-theme-name-free.sh;
    };

    checkThemeName = name:
    ''
      ${checkThemeNameScript}/bin/checkThemeName ${name}
    '';

    script = pkgs.writeScript "kate-check" (checkThemeName cfg.editor.theme.name);
```

Maybe it would be nicer to do this with just 1-2 definitions? My attempt would be:

```nix
    checkThemeNameScript = pkgs.writeShellApplication {
      name = "checkThemeName";
      runtimeInputs = with pkgs; [ jq ];
      text = builtins.readFile ./check-theme-name-free.sh;
    };

    script = pkgs.writeScript "kate-check" "${checkThemeNameScript}/bin/checkThemeName ${name}";
```

What do you think?